src/NuGet.Clients/NuGet.PackageManagement.UI/DeepLinkURIParser.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGet.PackageManagement.UI
+{
+    public static class DeepLinkURIParser
+    {
+        public static NuGetPackageDetails GetNuGetPackageDetails(string packageLink)
+        {
+            if (IsPackageURIValid(packageLink))
+            {
+                var packageInfo = packageLink.Split('/');
+
+                string packageName = packageInfo[3];
+                string versionNumber = packageInfo[4];
+
+                return new NuGetPackageDetails(packageName, versionNumber);
+            }
+            return null;
+        }
+        private static bool IsPackageURIValid(string packageLink)
+        {
+            if (packageLink == null)
+            {
+                return false;
+            }
+
+            var protocol = "vsph://";
+            var domain = "OpenPackageDetails/";
```

Comment:
This discussion belongs in the spec document, but considering that's not signed off before this PR was created:

Why these values? Does Visual Studio force us to use `vsph`?

As I wrote in the spec reivew, this scheme sounds awfully generic for any package manager, but Visual Studio supports NodeJS, Python and other projects that have their own package managers, so if we're going to use such generic names, we should provide extensibility for other project systems to use their own package manager UI, or we should make the URL format specific to NuGet.